pkg/volume/csi/csi_mounter.go

```diff
@@ -302,20 +302,13 @@ func getVolAttribsFromSpec(spec *volume.Spec) (map[string]string, error) {
 	if spec == nil {
 		return nil, errors.New("missing volume spec")
 	}
-	annotations := spec.PersistentVolume.GetAnnotations()
-	if annotations == nil {
-		return nil, nil // no annotations found
-	}
-	jsonAttribs := annotations[csiVolAttribsAnnotationKey]
-	if jsonAttribs == "" {
-		return nil, nil // csi annotation not found
-	}
-	attribs := map[string]string{}
-	if err := json.Unmarshal([]byte(jsonAttribs), &attribs); err != nil {
-		glog.Error(log("error parsing csi PV.Annotation [%s]=%s: %v", csiVolAttribsAnnotationKey, jsonAttribs, err))
-		return nil, err
+
```

Comment:
Function `getVolAttribsFromSpec()` found [here]( https://github.com/musse/kubernetes/blob/566ffc53a9b8bf9860b3b76f366f8150b90e3b4c/pkg/volume/csi/csi_mounter.go#L157) . Variable `csiSource`, declared [here](https://github.com/musse/kubernetes/blob/566ffc53a9b8bf9860b3b76f366f8150b90e3b4c/pkg/volume/csi/csi_mounter.go#L113), should already have the attributes (which is what the function is doing again).

Reply:
Done, let me know if the change is not what you expected.